modules/openapi-generator/src/main/resources/kotlin-multiplatform-client/common/main/infrastructure/ApiClientBase.kt.mustache

```diff
@@ -137,13 +135,13 @@ public open class ApiClientBase {
     }
 
     protected suspend fun jsonRequest(requestConfig: RequestConfig, body: Any? = null, authNames: List<String>): HttpResponse {
-        val contentType = (requestConfig.headers[HttpHeaders.ContentType]?.let { ContentType.parse(it) }
-                ?: ContentType.Application.Json)
-        return if (body != null) request(requestConfig, serializer.write(body, contentType), authNames)
+        if (HttpHeaders.ContentType !in requestConfig.headers)
+        requestConfig.headers[HttpHeaders.ContentType] = ContentType.Application.Json.toString()
+        return if (body != null) request(requestConfig, body, authNames)
```

Comment:
I think form based submission is also possible

Reply:
I think form and multipart go through  'urlEncodedFormRequest' and 'multipartFormRequest' respectively 
and requests with no body usually go to 'request' directly.

Reply:
Oh yeah you're right it's even called `jsonRequest`. It'll all go away anyway, so it should be fine